samples/go/nomdex/vrange.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/attic-labs/noms/go/types"
+)
+
+type ventry struct {
+	v    types.Value
+	incl bool
+	inf  int8
+}
+
+type vrange struct {
+	lower ventry
+	upper ventry
+}
+
+type vrangeslice []vrange
```

Comment:
It would be worth adding a comment saying that this sorts by the lower bound, since it's non-obvious how to sort ranges.